Libraries/LibGfx/ImageDecoder.cpp

```diff
@@ -40,7 +41,9 @@ ImageDecoder::ImageDecoder(const u8* data, size_t size)
     if (m_plugin->sniff())
         return;
 
-    m_plugin = nullptr;
+    m_plugin = make<JPGImageDecoderPlugin>(data, size);
+    if (m_plugin->sniff())
+        return;
 }
```

Comment:
I think you should keep the "m_plugin = nullptr;" part, and make it so `sniff()` actually does stuff (see below).
```suggestion

        m_plugin = nullptr;
        return;
}
```